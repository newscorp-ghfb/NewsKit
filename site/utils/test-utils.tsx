/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme: any = renderToFragmentWithThemeFactory(
  docsThemeLight,
);

export const mockNewsKitComponents = (...components: string[]) => () => ({
  ...(jest.requireActual('newskit') as any),
  ...components.reduce((acc, comp) => {
    acc[comp] = ({children, ...props}: any) => (
      <div
        data-comp={`NewsKit ${comp}`}
        data-testid={comp}
        data-props={JSON.stringify(props, null, 2)}
      >
        {children}
      </div>
    );
    return acc;
  }, {} as Record<string, Function>),
});
