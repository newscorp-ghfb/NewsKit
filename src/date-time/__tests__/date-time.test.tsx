import {renderToFragmentWithTheme} from '../../test/test-utils';
import {DateTime} from '..';

const date = '2019-01-25T15:32:00';

describe('DateTime', () => {
  test('renders with fallback formatting', () => {
    const fragment = renderToFragmentWithTheme(DateTime, {
      date,
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with additional HTML attributes', () => {
    const fragment = renderToFragmentWithTheme(DateTime, {
      date,
      id: 'date-time-id',
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with provided date, format, prefix and suffix', () => {
    const fragment = renderToFragmentWithTheme(DateTime, {
      date,
      prefix: 'Updated: ',
      dateFormat: 'd MMM yyyy, H:mm',
      suffix: ', The Times',
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with timestamp', () => {
    const timestamp = 1679993908342;
    const fragment = renderToFragmentWithTheme(DateTime, {
      date: timestamp,
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with overrides', () => {
    const fragment = renderToFragmentWithTheme(DateTime, {
      date: '2017-01-01T04:32:00.000Z',
      dateFormat: 'd MMM yyyy, H:mm',
      suffix: ', The Times',
      prefix: 'Updated:',
      overrides: {
        typographyPreset: 'utilityLabel010',
        stylePreset: 'articleHeadlineKicker',
        prefix: {
          typographyPreset: 'utilityLabel020',
          stylePreset: 'articleHeadlineKicker',
        },
        suffix: {
          typographyPreset: 'utilityLabel020',
          stylePreset: 'articleHeadlineKicker',
        },
      },
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with logical props overrides', () => {
    const fragment = renderToFragmentWithTheme(DateTime, {
      date: '2017-01-01T04:32:00.000Z',
      dateFormat: 'd MMM yyyy, H:mm',
      suffix: ', The Times',
      prefix: 'Updated:',
      overrides: {
        marginInline: '30px',
        paddingInline: '30px',
      },
    });
    expect(fragment).toMatchSnapshot();
  });

  test('renders with children', () => {
    const fragment = renderToFragmentWithTheme(DateTime, {
      date,
      children: '2 years ago',
    });
    expect(fragment).toMatchSnapshot();
  });
});
