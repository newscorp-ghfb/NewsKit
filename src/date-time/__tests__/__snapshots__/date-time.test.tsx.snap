// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTime renders with fallback formatting 1`] = `
<DocumentFragment>
  .emotion-1 {
  margin: 0;
  color: #5A6A79;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
  display: inline-block;
  white-space: pre;
}

.emotion-1 svg {
  fill: #5A6A79;
}

.emotion-1::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-1::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

.emotion-1:before,
.emotion-1:after {
  white-space: normal;
}

<time
    class="emotion-0"
  >
    <span
      class="emotion-1"
    >
      Mock Date 
    </span>
  </time>
</DocumentFragment>
`;

exports[`DateTime renders with logical props overrides 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin-inline: 30px;
  padding-inline: 30px;
}

.emotion-1 {
  margin: 0;
  color: #5A6A79;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
  display: inline-block;
  white-space: pre;
}

.emotion-1 svg {
  fill: #5A6A79;
}

.emotion-1::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-1::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

.emotion-1:before,
.emotion-1:after {
  white-space: normal;
}

<time
    class="emotion-0"
  >
    <span
      class="emotion-1"
    >
      Updated: 
    </span>
    <span
      class="emotion-1"
    >
      Mock Date, 
    </span>
    <span
      class="emotion-1"
    >
      , The Times
    </span>
  </time>
</DocumentFragment>
`;

exports[`DateTime renders with overrides 1`] = `
<DocumentFragment>
  .emotion-1 {
  margin: 0;
  font-family: "Poppins",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
  display: inline-block;
  white-space: pre;
}

.emotion-1::before {
  content: '';
  margin-bottom: -0.403em;
  display: block;
}

.emotion-1::after {
  content: '';
  margin-top: -0.4em;
  display: block;
}

.emotion-1:before,
.emotion-1:after {
  white-space: normal;
}

.emotion-2 {
  margin: 0;
  font-family: "Poppins",sans-serif;
  font-size: 12px;
  line-height: 18px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
  display: inline-block;
  white-space: pre;
}

.emotion-2::before {
  content: '';
  margin-bottom: -0.403em;
  display: block;
}

.emotion-2::after {
  content: '';
  margin-top: -0.4em;
  display: block;
}

.emotion-2:before,
.emotion-2:after {
  white-space: normal;
}

<time
    class="emotion-0"
  >
    <span
      class="emotion-1"
    >
      Updated: 
    </span>
    <span
      class="emotion-2"
    >
      Mock Date, 
    </span>
    <span
      class="emotion-1"
    >
      , The Times
    </span>
  </time>
</DocumentFragment>
`;

exports[`DateTime renders with provided date, format, prefix and suffix 1`] = `
<DocumentFragment>
  .emotion-1 {
  margin: 0;
  color: #5A6A79;
  font-family: "DM Sans",sans-serif;
  font-size: 14px;
  line-height: 21px;
  font-weight: 500;
  letter-spacing: 0;
  padding: 0.5px 0px;
  display: inline-block;
  white-space: pre;
}

.emotion-1 svg {
  fill: #5A6A79;
}

.emotion-1::before {
  content: '';
  margin-bottom: -0.391em;
  display: block;
}

.emotion-1::after {
  content: '';
  margin-top: -0.409em;
  display: block;
}

.emotion-1:before,
.emotion-1:after {
  white-space: normal;
}

<time
    class="emotion-0"
  >
    <span
      class="emotion-1"
    >
      Updated:  
    </span>
    <span
      class="emotion-1"
    >
      Mock Date, 
    </span>
    <span
      class="emotion-1"
    >
      , The Times
    </span>
  </time>
</DocumentFragment>
`;
